embedchain/embedchain.py

```diff
@@ -60,18 +61,19 @@ def _get_loader(self, data_type):
             'pdf_file': PdfFileLoader(),
             'web_page': WebPageLoader(),
             'qna_pair': LocalQnaPairLoader(),
-            'text': LocalTextLoader(),
+            'text': LocalTextLoader() if data_type == 'text' and os.path.isfile(url) else LocalTextLoader(),
         }
         if data_type in loaders:
             return loaders[data_type]
         else:
             raise ValueError(f"Unsupported data type: {data_type}")
 
-    def _get_chunker(self, data_type):
+    def _get_chunker(self, data_type, url=None):
```

Comment:
no need for extra `url`. We can let the `chunker` (ie. `TextChunker`) determine, when it is called, if the url is a local file and act accordingly.

Revisit other existing changes to adjust to this change.